octobot_backtesting/importers/exchanges/exchange_importer.py

```diff
@@ -102,116 +101,129 @@ async def _init_available_data_types(self):
 
-    def __get_operations_from_timestamps(self, superior_timestamp, inferior_timestamp):
-        operations: list = []
-        timestamps: list = []
-        if superior_timestamp != -1:
-            timestamps.append(str(superior_timestamp))
-            operations.append(enums.DataBaseOperations.INF_EQUALS.value)
-        if inferior_timestamp != -1:
-            timestamps.append(str(inferior_timestamp))
-            operations.append(enums.DataBaseOperations.SUP_EQUALS.value)
-
-        return timestamps, operations
-
-    def import_ohlcvs(self, ohlcvs):
-        for i, val in enumerate(ohlcvs):
-            ohlcvs[i] = list(val)
-            # ohlcvs[i][-2] = TimeFrames(ohlcvs[i][-2])
-            ohlcvs[i][-1] = json.loads(ohlcvs[i][-1])
-        return ohlcvs
-
     async def get_ohlcv(self, exchange_name=None, symbol=None,
                         time_frame=common_enums.TimeFrames.ONE_HOUR,
                         limit=data.DataBase.DEFAULT_SIZE):
-        return self.import_ohlcvs(await self.database.select(enums.ExchangeDataTables.OHLCV, size=limit,
-                                                             exchange_name=exchange_name, symbol=symbol,
-                                                             time_frame=time_frame.value))
+        return importers.import_ohlcvs(await self.database.select(enums.ExchangeDataTables.OHLCV, size=limit,
+                                                                  exchange_name=exchange_name, symbol=symbol,
+                                                                  time_frame=time_frame.value))
 
     async def get_ohlcv_from_timestamps(self, exchange_name=None, symbol=None,
                                         time_frame=common_enums.TimeFrames.ONE_HOUR,
                                         limit=data.DataBase.DEFAULT_SIZE,
                                         inferior_timestamp=-1, superior_timestamp=-1) -> list:
-        timestamps, operations = self.__get_operations_from_timestamps(superior_timestamp, inferior_timestamp)
-        return self.import_ohlcvs(await self.database.select_from_timestamp(enums.ExchangeDataTables.OHLCV, size=limit,
-                                                                            exchange_name=exchange_name, symbol=symbol,
-                                                                            time_frame=time_frame.value,
-                                                                            timestamps=timestamps,
-                                                                            operations=operations))
-
-    def import_tickers(self, tickers):
-        for ticker in tickers:
-            ticker[-1] = json.loads(ticker[-1])
-        return tickers
+        try:
+            self.cache[exchange_name][symbol][time_frame]
+        except KeyError:
+            cache = sorted(await self.get_ohlcv(exchange_name, symbol, time_frame, -1), key=lambda x: x[0])
```

Comment:
I don't think this is the proper place to manage cache

Reply:
I think all database client should have an abstract access to it. It should be transparent to them the cache usage.

Reply:
`ExchangeImporter` is now using a `ChronologicalReadDatabaseCache` (https://github.com/Drakkar-Software/OctoBot-Commons/pull/210/commits/428c7f7ec847374880c6f897210482c8b830e074#diff-bfdbdf813648264b818b4d67937d50bb5368a412ba8fff3e443b4afc6a91c72cR18) for all of its timestamp selects